go/tools/builders/nogo_main.go

```diff
@@ -202,6 +202,55 @@ func checkPackage(analyzers []*analysis.Analyzer, packagePath string, packageFil
 		act.pkg = pkg
 	}
 
+	// Process nolint directives similar to golangci-lint.
+	for _, f := range pkg.syntax {
+		// First, gather all comments and apply a range to the entire comment block.
+		// This will cover inline comments and allow us to detect comments above a
+		// node later.
+		for _, group := range f.Comments {
+			for _, comm := range group.List {
+				linters, ok := parseNolint(comm.Text)
+				if !ok {
+					continue
+				}
+				nl := &Range{
+					from: pkg.fset.Position(group.Pos()),
+					to:   pkg.fset.Position(group.End()).Line,
+				}
+				for analyzer, act := range actions {
+					if linters.Contains(analyzer.Name) {
```

Comment:
could `linters` be nil here?

Reply:
Yes, to indicate _all_ linters. `Contains` handles that case and always returns true.